File: JS/BOJ-S3-9375-Fashion_King_Shin_Haebin.js
'use strict';

const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : '../input.txt';
let input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split('\n')
  .map((v) => v.trimEnd());

let output = [];
const solution = (input) => {
  let t = +input[0];
  let idx = 1;
  while (t--) {
    let res = 1;
    const n = +input[idx];
    const m = new Map();
    for (let i = idx + 1; i <= idx + n; ++i) {
      const [a, b] = input[i].split(' ');
      if (m.has(b)) {
        m.set(b, m.get(b)++);
      } else {
        m.set(b, 1);
      }
    }
    idx += n + 1;
    m.forEach((v) => {
      res *= v + 1;
    });
    output.push(res - 1);
  }
};

solution(input);
// fs.writeFileSync('/dev/stdout', output.join('\n').trimEnd());
fs.writeFileSync(
  '../output.txt',
  output.join('\n').trimEnd(),
  (err) => console.error
);

File: CPP/BOJ-S3-9375-Fashion_King_Shin_Haebin.cpp
#include <bits/stdc++.h>
using namespace std;

int t, n;
string a, b;

int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  // freopen("input.txt", "r", stdin);
  // freopen("output.txt", "w", stdout);

  cin >> t;
  while (t--)
  {
    unordered_map<string, int> _map;
    cin >> n;
    for (int i = 0; i < n; ++i)
    {
      cin >> a >> b;
      _map[b]++;
    }
    // 경우의 수는 수가 커질 수 있기 때문에 long long 선언이 안전하다.
    long long ret = 1;
    for (auto c : _map)
    {
      ret *= ((long long)c.second + 1);
    }
    ret--;
    cout << ret << '\n';
  }

  return 0;
}